cache/contenthash/checksum.go

```diff
@@ -862,6 +862,10 @@ func (cc *cacheContext) scanChecksum(ctx context.Context, m *mount, p string, fo
 }
 
 func (cc *cacheContext) checksum(ctx context.Context, root *iradix.Node[*CacheRecord], txn *iradix.Txn[*CacheRecord], m *mount, k []byte, followTrailing bool) (*CacheRecord, bool, error) {
+	// only applies for Windows, it's a no-op on non-Windows.
+	enableProcessPrivileges()
```

Comment:
shouldn't this be in a more top-level function. This one gets called recursively I think.

Reply:
Let me crosscheck.

Reply:
You right, it's recursive here. I've taken it up to a more top-level function now.